tokio/tests/sync_mpsc.rs

```diff
@@ -545,6 +545,56 @@ async fn try_reserve_fails() {
     let _permit = tx.try_reserve().unwrap();
 }
 
+#[maybe_tokio_test]
+async fn try_reserve_many_fails() {
+    for i in 4..20 {
+        let (tx, mut rx) = mpsc::channel(i);
+
+        let mut permit1 = assert_ok!(tx.try_reserve_many(i - 2));
+
+        // This should fail, there is only two remaining permits
+        match assert_err!(tx.try_reserve_many(3)) {
+            TrySendError::Full(()) => {}
+            _ => panic!(),
+        }
+
+        permit1.next().unwrap().send("foo");
+        permit1.next().unwrap().send("foo");
+
+        assert_eq!(rx.recv().await, Some("foo"));
+        assert_eq!(rx.recv().await, Some("foo"));
+
+        // There are now 4 remaining permits because of the 2 sends/recv
+        let permit2 = assert_ok!(tx.try_reserve_many(4));
+
+        // Dropping permit iterator releases the remaining slots.
+        drop(permit1);
+        drop(permit2);
+
+        // It is possible to reserve all the permits
+        assert_ok!(tx.try_reserve_many(i));
+
+        // This should fail because the channel is closed
+        drop(rx);
+        match assert_err!(tx.try_reserve_many(3)) {
+            TrySendError::Closed(()) => {}
+            _ => panic!(),
+        };
+    }
+}
+
+#[maybe_tokio_test]
+async fn reserve_many_and_send() {
+    let (tx, mut rx) = mpsc::channel(100);
+    for i in 1..100 {
```

Comment:
This test is really nice because it is so small. 
```suggestion
    for i in 0..100 {
```